src/SFA.DAS.AssessorService.Data/RegisterQueryRepository.cs

```diff
@@ -440,5 +439,25 @@ public async Task<EpaContact> GetContactBySignInId(string signinId)
                 return deliveryAreas;
             }
         }
+
+
+        public async Task<string> GetEpaOrgIdByEndPointAssessmentName(string name)
+        {
+            using (var connection = new SqlConnection(_configuration.SqlConnectionString))
+            {
+                name = name.Replace(" ", "");
+                if (connection.State != ConnectionState.Open)
+                    await connection.OpenAsync();
+                var sqlForMainDetails =
+                    "SELECT  O.EndPointAssessorOrganisationId " +
+                    " FROM [Organisations] O " 
+                    + "WHERE replace(o.EndPointAssessorName, ' ','') like @name "
+                    + "OR replace(JSON_VALUE(o.[OrganisationData], '$.TradingName'), ' ','') like @name "
+                    + "OR replace(JSON_VALUE(o.[OrganisationData], '$.LegalName'), ' ','') like @name ";
+                var orgs = await connection.QueryAsync<string>(sqlForMainDetails, new { name });
+                var org = orgs.FirstOrDefault();
+                return org;
+            }
+        }
```

Comment:
Why would we need to search Organisations by name?  This seems to be used by CreateEpaOrganisationHandler to check if an org already exists, so why not check on ukprn?  Or is there a case where we wouldn't have the ukprn at this point?